pkg/controllers/nodeclaim/disruption/controller.go

```diff
@@ -115,8 +117,9 @@ func (c *Controller) Reconcile(ctx context.Context, nodeClaim *v1beta1.NodeClaim
 
 func (c *Controller) Register(_ context.Context, m manager.Manager) error {
 	builder := controllerruntime.NewControllerManagedBy(m)
-	for _, ncGVK := range c.cloudProvider.GetSupportedNodeClasses() {
+	for _, supportedNC := range c.cloudProvider.GetSupportedNodeClasses() {
 		nodeclass := &unstructured.Unstructured{}
```

Comment:
Do you even need to do unstructured if you have the `status.Object`? I think you can just directly use the types now